plugins/services/src/js/components/ServiceBreadcrumbs.js

```diff
@@ -45,6 +44,22 @@ class ServiceBreadcrumbs extends React.Component {
     global.addEventListener("resize", this.handleViewportResize);
   }
 
+  shouldComponentUpdate(nextProps) {
+    const hasServiceIDChanged = this.props.serviceID !== nextProps.serviceID;
+    const hasTaskIDChanged = this.props.taskID !== nextProps.taskID;
+    const hasTaskNameChanged = this.props.taskName !== nextProps.taskName;
+    const hasExtraNotChanged =
+      this.props.extra === nextProps.extra ||
+      deepEqual(this.props.extra, nextProps.extra);
```

Comment:
Why the equality check before the deepEqual?

Reply:
As deepEqual is not a lightweight function I thought we would rather not call that function in obvious cases. 

Unfortunately, we have to call it since nodes are never equal.

Also please consider that this function is called every 2 seconds. And extra is undefined in most cases.

Reply:
Okay, got you there 👍

Reply:
@Poltergeist could you please file a JIRA to adjust deepEqual so that we don't need to  first check whether the two options are the same here?

Reply:
Maybe we can do that on master, but let's stay with `deep-equal` on the legacy branches 👍

Reply:
+1 Michaels suggestion

Reply:
I like the idea too, I just would defer that for a later time, given that we want quickly ship these performance improvements.

Reply:
After offline discussion with philipp: replacing deep-equal wit isEqual should be its own task. AFAIK philipp is checking right now if it is significantly faster 👍

Reply:
I looked into the performance, and yes lo-dash isEqual is seemed to be a lot faster. But my Opinion on this is that introducing a new module is not better solving the issue. And it is a scope creep. We have the deep-equal module already installed and use it in multiple places in the code base. So I only adopted that pattern. I created an issue for the next sprint to replace deep-equal with lodash.isEqual [DCOS-37562](https://jira.mesosphere.com/browse/DCOS-37562) I hope that this approach is resolving this issue @orlandohohmeier @weblancaster